capi-configuration/3_upgrades.md

```diff
@@ -0,0 +1,52 @@
+# Handling workload cluster upgrades
+
+## Problem Statement
+
+We want to ...
+1. Make all changes transparent for the user before application
+2. Continue supporting the current workflow of upgrading a cluster (Through GS staff, customers and automation)
+3. Integrate the process into the proposed gitops-management
+4. Utilize the managed-configmap approach to ensure clusters adhere to the definition in their release at all times
+5. Ensure that an upgrade to a cluster is decoupled from upgrades of other components
+6. Allow easy integration into other pipelines and formats users might want to utilize for cluster management.
+
+## Triggering an Upgrade
+
+Our current method to trigger an upgrade is to change the release-version label on the `Cluster` resource. This can be done by GS staff, customers and automation through various interfaces.
+This change will trigger the actual upgrade by letting a newer set of components manage the cluster.
+However, when introducing the [new structure for releases](0_capi-releases), this process will be different. The release will no longer be tied to the components.
```

Comment:
I think it would be valuable to explain here why changing the release label will not work in the future. It is just completely incompatible with a gitops approach to cluster management.